data-prepper-api/src/main/java/com/amazon/dataprepper/model/buffer/AbstractBuffer.java

```diff
@@ -17,10 +17,12 @@
 public abstract class AbstractBuffer<T extends Record<?>> implements Buffer<T> {
     protected final PluginMetrics pluginMetrics;
     private final Counter recordsWrittenCounter;
-    private final Counter recordsReadCounter;
+    private final Counter recordsInflightCounter;
+    private final Counter recordsProcessedCounter;
```

Comment:
I think the name of this metric is correct, it is a count of the records processed by this pipeline. However, it would be nice if this could be a pipeline level metric rather than a buffer level metric. 

pipelineName.buffer.recordsProcessed seems to imply that it is a count of the records processed by that buffer, rather than the records processed by the pipeline.

recordsRead, recordsWritten, read/write latency, and inflight are all specific to the buffer. Records processed refers to records processed by the pipeline

Reply:
In our case, although inflight could be specific to buffer, it still needs to be updated when recordsProcessed is updated. So we need either (1) expose recordsInflightCounter to the popeline or (2) make recordsInflightCounter a pipeline level metric.

Reply:
Or what if we still keep it as a buffer level metric but change the name into pipelineName.buffer.recordsChecked?

Reply:
Do we have a conclusion on this discussion? Asking because I missed the offline conversation..

Reply:
Yes. The solution is to make the prefix of the metric name reflect the pipeline level, i.e. `pipelineName.recordsProcessed` instead of `pipelineName.bounded_buffer.recordsProcessed`